backend/server.py

```diff
@@ -0,0 +1,284 @@
+from ldm.gfpgan.gfpgan_tools import gfpgan_model_exists
+from ldm.dream.pngwriter import PngWriter
+from ldm.generate import Generate
+from flask_socketio import SocketIO
+from flask import Flask, send_from_directory, url_for, jsonify
+import transformers
+import json
+import base64
+import mimetypes
+import os
+import sys
+import signal
+import time
+import eventlet
+import traceback
+from threading import Event
+from enum import Enum
+
+from pathlib import Path
+from pytorch_lightning import logging
+from parse_seed_weights import parse_seed_weights
+
+# fix missing mimetypes on windows due to registry wonkiness
+import mimetypes
+mimetypes.add_type('application/javascript', '.js')
+mimetypes.add_type('text/css', '.css')
+
+# Host and port to serve on
+host = 'localhost'
+port = 9090
+
+
+"""
+Additional CORS origins to this list.
+Useful if allowing other machines on your network to use the app.
+
+Example:
+additional_allowed_origins = ["192.168.1.8","192.168.1.24"]
+"""
+additional_allowed_origins = []
+
+
+def build_cors_allowed_origins(additional_allowed_origins):
+    cors_allowed_origins = [f"http://{host}:{port}"]
```

Comment:
This entire body can be replaced with `    return [f"http://{origin}:{port}" for origin in additional_allowed_origins]`.

Note that currently it is returning `http://host:port` (which IIUC is unnecessary), and IPs, which I don't *think* is correct (but frankly the docs aren't real clear here). My assumption above is that an "origin" must be of the form `http://address:port`

Reply:
That would miss the initial `cors_allowed_origins` value, though.

I haven't python'd in a long time and just remembered you can simply `+` arrays.

```suggestion
    cors_allowed_origins = [f"http://{host}:{port}"] + additional_allowed_origins
```

Reply:
> That would miss the initial cors_allowed_origins value, though

Exactly :D

It didn't make sense to me that you'd have to unblock the *default* as well, but I could be wrong about that.

Reply:
Ooooh. Gotcha. I think you do have to unblock the default actually. Anyways, with @Kyle0654 's *far* more sophisticated API incoming, I'm just going to leave my "API" in it's technically-ain't-broke state until migrating to their API.